server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2911,9 +2924,42 @@ public UserVm updateVirtualMachine(long id, String displayName, String group, Bo
             }
         }
 
+        if (!StringUtils.isEmpty(hostName)) {
+            vm.setHostName(hostName);
+            try {
+                updateDns(vm);
+            } catch (CloudRuntimeException e) {
+                throw new CloudRuntimeException(String.format("Failed to update hostname of VM %s to %s", vm.getInstanceName(), vm.getHostName()));
+            }
+        }
+
         return _vmDao.findById(id);
     }
 
+    private void updateDns(UserVmVO vm) {
```

Comment:
@Pearl1594 
do we need to update dns if vm is Stopped ?

Reply:
It doesn't cause any issues - but you are right it's worth skipping it for a shutoff VM.